src/objects.rs

```diff
@@ -691,6 +691,44 @@ impl GetObjectType for VehicleJourney {
     }
 }
 
+#[derive(Debug)]
+pub enum StopTimeError {
+    DuplicateStopSequence { vj_id: String },
+    IncoherentStopTimes { vj_id: String },
+}
+
+impl std::fmt::Display for StopTimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StopTimeError::DuplicateStopSequence { vj_id } => {
+                write!(f, "duplicate stop_sequence for the trip '{}'", vj_id)
+            }
+            StopTimeError::IncoherentStopTimes { vj_id } => {
+                write!(f, "incoherent stop times for the trip '{}'", vj_id)
+            }
+        }
+    }
+}
+
+impl VehicleJourney {
+    pub fn sort_and_check_stop_times(&mut self) -> Result<(), StopTimeError> {
+        self.stop_times.sort_unstable_by_key(|st| st.sequence);
+        for (curr_st, next_st) in self.stop_times.iter().zip(self.stop_times.iter().skip(1)) {
```

Comment:
I think you can use `.windows()` function for the same effect.

```suggestion
        for [curr_st, next_st] in self.stop_times.windows(2) {
```

Reply:
I didn’t know it but "refutable pattern in `for` loop binding: `&[]`, `&[_]` and `&[_, _, _, ..]` not covered"

Reply:
Then you can do it like this in this case.

```suggestion
        for window in self.stop_times.windows(2) {
            let curr_st = &window[0];
            let next_st = &window[1];
```

Reply:
done in d03225efc5decf59f273b03e368313e49e03f244